cmake/app/boilerplate.cmake

```diff
@@ -87,6 +87,12 @@ set(__build_dir ${CMAKE_CURRENT_BINARY_DIR}/zephyr)
 
 set(PROJECT_BINARY_DIR ${__build_dir})
 
+define_property(GLOBAL PROPERTY ${IMAGE}PROJECT_BINARY_DIR
+  BRIEF_DOCS "Build directory (PROJECT_BINARY_DIR) for the ${IMAGE} image."
+  FULL_DOCS "To be used to access e.g. this image's hex file."
+  )
+set_property(GLOBAL PROPERTY ${IMAGE}PROJECT_BINARY_DIR ${PROJECT_BINARY_DIR})
```

Comment:
Introducing ${IMAGE}PROJECT_BINARY_DIR is sane.

From the CMakeCache.txt I see that ${project}_BINARY_DIR is already available. But that isn't quite what we want unfortunately.